### ISSUE TITLE
Edge case: custom exception types

### ISSUE THREAD

**Issue by ewels** [ewels/rich-click#27]
Came across an edge case issue in DDS:

```python
  File "/home/sjunnebo/.conda/envs/dds/lib/python3.8/site-packages/rich_click/rich_click.py", line 449, in rich_format_error
    if self.ctx is not None:
AttributeError: 'DDSCLIException' object has no attribute 'ctx'
```

This seems to be because there is a [custom exception type](https://github.com/ScilifelabDataCentre/dds_cli/blob/19f03be02995eed63026e91c69f1fa71135ee074/dds_cli/exceptions.py#L30) based on Click which is being used. I can't imagine that this comes up a lot, but it would be good to cope with missing `ctx` in the error handling..

**Comment by ewels** [comment]
Difficult to replicate / test, but hopefully this should now be fixed.

**Comment by ewels** [comment]
Published in v1.2.1 https://pypi.org/project/rich-click/1.2.1/
